topydo/commands/TopCommand.py

```diff
@@ -0,0 +1,105 @@
+# Topydo - A todo.txt client written in Python.
+# Copyright (C) 2014 - 2015 Bram Schoenmakers <me@bramschoenmakers.nl>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from topydo.lib.ExpressionCommand import ExpressionCommand
+from topydo.lib.Config import config
+from topydo.lib.PrettyPrinter import pretty_printer_factory
+from topydo.lib.PrettyPrinterFilter import (
+    PrettyPrinterIndentFilter,
+    PrettyPrinterHideTagFilter,
+    PrettyPrinterBasicPriorityFilter,
+    PrettyPrinterHumanDatesFilter
+)
+from topydo.lib.Utils import get_terminal_size
+
+
+class TopCommand(ExpressionCommand):
```

Comment:
The two are fairly similar, but (almost) every function (\__init\__, _process_flags, _print, usage, help) is actually slightly different between the two.

Because they run different code, the tests for `top` are basically the same as `list`'s to make sure that if either one breaks it is caught right away.

I'm open to subclassing `ListCommand`, I'm just not sure how to do it in a way that makes sense.